sysutils/apcupsd/src/opnsense/mvc/app/models/OPNsense/Apcupsd/Apcupsd.xml

```diff
@@ -0,0 +1,163 @@
+<model>
+    <mount>//OPNsense/apcupsd</mount>
+    <version>0.0.2</version>
+    <description>APC UPS configuration</description>
+    <items>
+        <!-- Container -->
+        <general>
+            <!-- field -->
+            <Enabled type="BooleanField">
+                <default>0</default>
+                <Required>Y</Required>
+            </Enabled>
+            <UPSName type="TextField">
+                <default>Firewall_UPS</default>
+                <Required>Y</Required>
+                <ValidationMessage>A name for the UPS is required.</ValidationMessage>
+            </UPSName>
+            <UPSCable type="OptionField">
+                <default>smart</default>
+                <Required>Y</Required>
+                <OptionValues>
+                    <option value="simple">Simple</option>
+                    <option value="smart">Smart</option>
+                    <option value="ether">Ether</option>
+                    <option value="usb">USB</option>
+                    <option value="940-0020B">940-0020B</option>
+                    <option value="940-0020C">940-0020C</option>
+                    <option value="940-0023A">940-0023A</option>
+                    <option value="940-0024B">940-0024B</option>
+                    <option value="940-0024C">940-0024C</option>
+                    <option value="940-0024G">940-0024G</option>
+                    <option value="940-0095A">940-0095A</option>
+                    <option value="940-0095B">940-0095B</option>
+                    <option value="940-0095C">940-0095C</option>
+                    <option value="940-0119A">940-0119A</option>
+                    <option value="940-0127A">940-0127A</option>
+                    <option value="940-0128A">940-0128A</option>
+                    <option value="940-0625A">940-0625A</option>
+                    <option value="940-1524C">940-1524C</option>
+                    <option value="MAM-04-02-2000">MAM-04-02-2000</option>
+                </OptionValues>
+            </UPSCable>
+            <UPSType type="OptionField">
+                <default>apcsmart</default>
+                <Required>Y</Required>
+                <OptionValues>
+                    <apcsmart>apcsmart</apcsmart>
+                    <usb>usb</usb>
+                    <net>net</net>
+                    <snmp>snmp</snmp>
+                    <netsnmp>netsnmp</netsnmp>
+                    <dumb>dumb</dumb>
+                    <pcnet>pcnet</pcnet>
+                    <modbus>modbus</modbus>
+                </OptionValues>
+            </UPSType>
+            <Device type="TextField">
+                <Required>N</Required>
```

Comment:
Why is this not required but has a default? Is there a special reason?

Reply:
> Why is this not required but has a default? Is there a special reason?

Reason is that the DEVICE can be empty. Depends on the UPSType.

See http://www.apcupsd.org/manual/manual.html#usb-configuration

"DEVICE device
Specify which device is used for UPS communications. 
For serial ports, it is usually something like /dev/ttyS0. 
For USB ports, you may leave the name of the device blank (no specification) and apcupsd will automatically search the standard locations for the UPS."

But for the user to see the syntax we decided to include a default for the most common one.
//Dan Lundqvist (mrzaz)

Reply:
Ok, then its fine

Reply:
I removed the default and suggested what to put in there in the help text, because I was thinking that leaving a device path in there may create confusion especially if that device path doesn't exist.

I can revert or change it again if you think it's better to leave a default there.